pygsti/data/dataset.py

```diff
@@ -1730,6 +1730,11 @@ def _add_raw_arrays(self, circuit, oli_array, time_array, rep_array,
                 # so we need to build this up for all existings sequences:
                 self._add_explicit_repetition_counts()
 
+        if rep_array is not None:
+            dtype_info = _np.finfo(rep_array.dtype)
+            near_zero = rep_array < 10**(-1.5*dtype_info.precision)
+            rep_array[near_zero] = 0
+
```

Comment:
Just to double-check, this was setting things that are numerically 0 to 0 so that line 1741 drops it if record_zero_counts is False? If so, can we add that as a comment? I don't work with the array mask syntax much and this took me a second.

Reply:
Yup! Comment added :)

Reply:
Thanks! Also, just a random sidenote: This looked much more natural to me in 515 where it was just array[array < 1e-10] = 0... I don't know why having the mask split out on its own line took me by surprise 😅